netstandard/tools/Trimming/NuGetPackageNode.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    /// <summary>
+    /// Metedata and information for a package listed in the lock file.
+    /// </summary>
+    internal sealed class NuGetPackageNode : IIsIncluded
+    {
+        private const string NuGetPackageDependencies = "dependencies";
+        private List<NuGetPackageNode> _dependencies = new List<NuGetPackageNode>();
+        private IEnumerable<string> _dependencyIds;
+
+        public NuGetPackageNode(string id, string version, IEnumerable<string> dependencyIds)
+        {
+            Id = id;
+            IsRuntimePackage = id.StartsWith("runtime.");
+            Version = version;
+            _dependencyIds = dependencyIds;
+        }
+
+        public string Id { get; }
+        private bool IsRuntimePackage { get; }
+        public bool IsIncluded { get; set; }
+        public string Version { get; }
+        public bool IsProject { get; }
+        public IEnumerable<NuGetPackageNode> Dependencies { get { return _dependencies; } }
+        public IList<FileNode> Files { get; } = new List<FileNode>();
+
+        public void PopulateDependencies(Dictionary<string, NuGetPackageNode> allPackages)
+        {
+            foreach(var dependencyId in _dependencyIds)
+            {
+                NuGetPackageNode dependencyNode;
+                if (!allPackages.TryGetValue(dependencyId, out dependencyNode))
+                {
+                    // package declared a dependency but NuGet was missing the dependent package
+                    // in the lock file.  This indicates a broken restore, but don't fail trimming
+                }
+                else
+                {
+                    _dependencies.Add(dependencyNode);
+
+                    // Runtime packages may be brought in by a file-based dependency,
```

Comment:
I see yes so they aren't depending on the runtime package just a file from the runtime package.